### PR TITLE
Refactors single verified list

### LINKED ISSUE #14691: Publisher verification lookup should use a singe list source
### Description
Currently the publisher verification check implemented in browser-laptop does a check for each individual publisher using the `/v3/publisher/identity` route. Instead, a single list of verified publishers should be downloaded and cached periodically so that the checks can be made against that resource instead.

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves #14691

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).
- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review). (Ask a Brave employee to help if you cannot access this document.)

## Test Plan:
test plan in https://github.com/brave/browser-laptop/pull/14708#issue-200508550

## Reviewer Checklist:

- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review) if one was not already requested.

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

File: app/browser/api/ledger.js [hunk 1]
```diff
@@ -2369,6 +2371,7 @@ const initialize = (state, paymentsEnabled) => {
     // First run
     module.exports.checkPublisherInfoUpdate(state)
     // Check again every other day
+    // TODO I think that we need to clear this interval if user turns payments off
```
Comment (NejcZdovc) [inline app/browser/api/ledger.js]: @ryanml

Comment (ryanml) [reply app/browser/api/ledger.js]: Yep, agreed. No purpose for at that point. List fetches will continue if they reenable

Comment (NejcZdovc) [reply app/browser/api/ledger.js]: done

**Comment by NejcZdovc** [comment]
master 7ebee8e
0.24 ab68730dc
0.23 6de29b126

**Pull request merged**